app/uk/gov/hmrc/bindingtariffclassification/model/reporting/CaseReport.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.bindingtariffclassification.model.reporting
+
+import play.api.mvc.QueryStringBindable
+import uk.gov.hmrc.bindingtariffclassification.model.reporting.CaseReportField.CaseReportField
+import uk.gov.hmrc.bindingtariffclassification.model.reporting.CaseReportGroup.CaseReportGroup
+
+case class CaseReport
+(
+  filter: CaseReportFilter,
+  group: CaseReportGroup,
+  field: CaseReportField
+)
+
+object CaseReport {
+  private val reportGroupKey = "report_group"
+  private val reportFieldKey = "report_field"
+
+  implicit def bindable(implicit stringBinder: QueryStringBindable[String],
+                        filterBinder: QueryStringBindable[CaseReportFilter]
+                     ): QueryStringBindable[CaseReport] = new QueryStringBindable[CaseReport] {
+    override def bind(key: String, requestParams: Map[String, Seq[String]]): Option[Either[String, CaseReport]] = {
+      import uk.gov.hmrc.bindingtariffclassification.model.utils.BinderUtil._
+      implicit val rp: Map[String, Seq[String]] = requestParams
+
+      val filter: CaseReportFilter = filterBinder.bind("", requestParams).get.right.get
```

Comment:
is get safe here?   No idea what filterBinder.bind returns

Reply:
If you have a look in the Filter binder, it always returns `Some(Right(...))` so it's safe